### PR TITLE
Electric Eel

### PR DESCRIPTION
**Pull request by Nibbl-z**
Awesome minigame. i love stealinig radioactive materials

### CODE REVIEW

File: src/main/java/xyz/devcmb/cmr/interfaces/scoreboards/minigames/ElectricEelScoreboard.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import org.bukkit.entity.Player;
+import org.bukkit.scoreboard.*;
+import xyz.devcmb.cmr.GameManager;
+import xyz.devcmb.cmr.interfaces.scoreboards.HandledScoreboard;
+import xyz.devcmb.cmr.minigames.ElectricEelController;
+import xyz.devcmb.cmr.utils.Utilities;
+
+public class ElectricEelScoreboard implements HandledScoreboard {
+    private final ElectricEelController electricEelController;
+    public ElectricEelScoreboard(ElectricEelController kc){
```
Comment (29cmb) [inline src/main/java/xyz/devcmb/cmr/interfaces/scoreboards/minigames/ElectricEelScoreboard.java:14]: In the future, please rename it from kc to the controller initials

File: src/main/java/xyz/devcmb/cmr/listeners/minigames/ElectricEelListeners.java [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+        double minY = Math.min(point1.getY(), point2.getY());
+        double minZ = Math.min(point1.getZ(), point2.getZ());
+        double maxX = Math.max(point1.getX(), point2.getX());
+        double maxY = Math.max(point1.getY(), point2.getY());
+        double maxZ = Math.max(point1.getZ(), point2.getZ());
+
+        return loc.getX() >= minX && loc.getX() <= maxX &&
+                loc.getY() >= minY && loc.getY() <= maxY &&
+                loc.getZ() >= minZ && loc.getZ() <= maxZ;
+    }
```
Comment (29cmb) [inline src/main/java/xyz/devcmb/cmr/listeners/minigames/ElectricEelListeners.java:86]: Please move this to the utilities class

File: src/main/java/xyz/devcmb/cmr/listeners/minigames/ElectricEelListeners.java [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+                }
+                electricEelController.blueUranium--;
+                meta.getPersistentDataContainer().set(key, PersistentDataType.STRING, "BLUE");
+            }
+
+            uranium.setItemMeta(meta);
+            event.getPlayer().getInventory().setItemInOffHand(uranium);
+
+            PotionEffect glowing = new PotionEffect(PotionEffectType.GLOWING, Integer.MAX_VALUE, 1, false, false);
+            player.addPotionEffect(glowing);
```
Comment (29cmb) [inline src/main/java/xyz/devcmb/cmr/listeners/minigames/ElectricEelListeners.java:145]: Use player.setGlowing instead

File: src/main/java/xyz/devcmb/cmr/listeners/minigames/ElectricEelListeners.java [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+
+            world.playSound(event.getBlock().getLocation(), Sound.BLOCK_BEACON_DEACTIVATE, 1, 1);
+        }
+
+        electricEelController.ResetBeams(electricEelLocation);
```
Comment (29cmb) [inline src/main/java/xyz/devcmb/cmr/listeners/minigames/ElectricEelListeners.java:150]: You're resetting the beams on every block, not just ones that need to reset them

File: src/main/java/xyz/devcmb/cmr/listeners/minigames/ElectricEelListeners.java [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+                electricEelController.redUranium++;
+            } else if (isWithin(event.getBlock().getLocation(), blueStorageFromLocation, blueStorageToLocation)) {
+                electricEelController.blueUranium++;
+            }
+
+            event.getPlayer().removePotionEffect(PotionEffectType.GLOWING);
+            world.playSound(event.getBlock().getLocation(), Sound.BLOCK_BEACON_ACTIVATE, 1, 1);
+        }
+
+        electricEelController.ResetBeams(electricEelLocation);
```
Comment (29cmb) [inline src/main/java/xyz/devcmb/cmr/listeners/minigames/ElectricEelListeners.java:179]: Same here

File: src/main/java/xyz/devcmb/cmr/minigames/ElectricEelController.java [hunk 6]
```diff
@@ -0,0 +1,486 @@
...
+    }
+
+    @Override
+    public String getName() {
+        return "Electric Eel";
+    }
+
+    @Override
+    public String getDescription() {
+        return "";
```
Comment (29cmb) [inline src/main/java/xyz/devcmb/cmr/minigames/ElectricEelController.java:484]: Buddy

**Review by 29cmb — CHANGES_REQUESTED** [review]
Why does it not have a description, also please look into the optimization issues

File: src/main/java/xyz/devcmb/cmr/minigames/ElectricEelController.java (hunk 6 above)
Comment (Nibbl-z) [reply src/main/java/xyz/devcmb/cmr/minigames/ElectricEelController.java:484]: SOWWY I FORGOT

**Comment by Nibbl-z** [comment]
Fixing time! :steamhappy:

**Comment by Nibbl-z** [comment]
image

**Review by 29cmb — APPROVED** [review]
Lgtm

**Pull request merged**